src/igvfd/types/file_set.py

```diff
@@ -30,6 +30,15 @@ def get_fileset_objs_from_input_file_sets(request, input_file_sets):
     return file_set_objs
 
 
+def get_file_objs_from_files(request, files):
+    '''Get file objects from an array of files'''
+    file_objs = []
+    if files is not None:
+        for file in files:
+            file_objs.append(request.embed(file, '@@object'))
```

Comment:
For the frame use '@@object?skip_calculated=true' instead. There is probably a reverse link or something else that is causing a recursion error.